source/flow.cpp

```diff
@@ -53,14 +54,14 @@ Time Flow::create_new_data_packet() {
 }
 
 Time Flow::put_data_to_device() {
-    m_src->enqueue_packet(m_sending_buffer.front());
+    m_src.lock()->enqueue_packet(m_sending_buffer.front());
```

Comment:
Please, check that device you use is not expired:
```suggestion
    if (m_src.expired()) {
        LOG_ERROR("Flow source was deleted; can not put data to it");
        return 0;
    }
    m_src.lock()->enqueue_packet(m_sending_buffer.front());
```